src/app/core/services/helper.service.spec.ts

```diff
@@ -146,4 +146,18 @@ describe('Service: Helper', () => {
     const result = helperService.capitalizeFirstLetterOfString(testString);
     expect(result).toEqual('');
   });
+
+  it('should return a url with parameters', () => {
+    const url = '/genes/ENSG123';
+    const expected = '/genes/ENSG123?model=abc';
+    const result = helperService.addUrlParam(url, 'model', 'abc');
+    expect(result).toEqual(expected);
+  });
+
+  it('should replace existing url parameters if the same one exists', () => {
+    const url = '/genes/ENSG123?model=abc';
+    const expected = '/genes/ENSG123?model=fff';
+    const result = helperService.addUrlParam(url, 'model', 'fff');
+    expect(result).toEqual(expected);
+  });
```

Comment:
Nice! Could you add another test that demonstrates the expected behavior when the URL contains an existing, different query parameter? For example, if the url is `/genes/ENSG123?existing=abc` and we call `helperService.addUrlParam(url, 'model', 'fff')`, what should the result be?

Reply:
I'll add the test to have it replace the url parameter